### PR TITLE
Removes the advanced shotgun ammo boxes

### PR DESCRIPTION
**Pull request by tf-4**
## About The Pull Request

- Removes the advanced shotgun ammo boxes
- Adds regular boxes for each special shotgun ammo type
- Lowers the price per shot to be equivalent to before
- Increases the number of boxes available (a little, could increase it more)
- Removes unused hunter ammo type (disabled for being extremely broken)
- Puts some of the description detail from the shotgun ammo onto the boxes

## How This Contributes To The Skyrat Roleplay Experience

These ammo boxes destroy half the downside of using a shotgun, that being slow, clunky reloads with limited ammo. They've been overpowered since they were first added.
Regular boxes of ammo still let you purchase the wacky (read: overpowered, but we'll address that another time) ammo types that we have, without nullifying a balance aspect of the weapon they're designed for.

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary>
  
ammo
crates
box inside

</details>

## Changelog
:cl:
balance: Removed the advanced shotgun ammo boxes
balance: Lowered the cost of shotgun ammunition per box in cargo, though it costs the same per shell
spellcheck: Added more detail to the shotgun ammo box descriptions
/:cl:

### CODE REVIEW

**Comment by Zonespace27** [comment]
The advanced shotgun boxes were one of those things that you'd overlook on paper but dear lord they were effective in game

**Comment by YakumoChen** [comment]
I used the shotgun, you know why?

File: modular_skyrat/modules/shotgunrebalance/code/ammobox.dm [hunk 1]
```diff
@@ -1,96 +1,154 @@
...
+		new /obj/item/ammo_casing/shotgun/incendiary(src)
+
+// CONFETTI
+/obj/item/storage/box/shotgun/confetti
+	name = "box of confetti shells"
+	desc = "A box of 12 gauge buckshot shells that have been filled to the brim with confetti."
+	icon_state = "confetti"
+
+/obj/item/storage/box/shotgun/confetti/PopulateContents()
+	for(var/i in 1 to 7)
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/shotgunrebalance/code/ammobox.dm]: Is there a reason why this `for(var/i in 1 to 7)` loop can't be made generic by using a variable that stands for the ammo type instead? Could just make it on the `/obj/item/storage/box/shotgun` and then have the `7` be a variable for how many bullets to spawn.

File: modular_skyrat/modules/shotgunrebalance/code/shotgun.dm [hunk 2]
```diff
@@ -18,10 +18,14 @@
...
-	desc = "A 12 gauge plastitanium slug purpose built to penetrate armored targets."
+	desc = "A 12 gauge plastitanium slug purpose-built to penetrate armored targets."
 	icon_state = "apshell"
 	projectile_type = /obj/projectile/bullet/shotgun_slug/pt20
-	custom_materials = list(/datum/material/iron=500,/datum/material/plasma=500,/datum/material/titanium=500)
+	custom_materials = list(
+		/datum/material/iron = 500,
+		/datum/material/plasma = 500,
+		/datum/material/titanium = 500,
+		)
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/shotgunrebalance/code/shotgun.dm]:
```suggestion
	)
```

File: modular_skyrat/modules/shotgunrebalance/code/shotgun.dm [hunk 3]
```diff
@@ -33,13 +37,26 @@
 	desc = "Radically Invasive Projectile Slug that is designed to cause massive damage against unarmored targets by embedding inside them."
 	icon_state = "ripshell"
 	projectile_type = /obj/projectile/bullet/shotgun_slug/rip
-	custom_materials = list(/datum/material/iron=500,/datum/material/plasma=500,/datum/material/diamond=500)
+	custom_materials = list(
+		/datum/material/iron = 500,
+		/datum/material/plasma = 500,
+		/datum/material/diamond = 500,
+		)
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/shotgunrebalance/code/shotgun.dm]:
```suggestion
	)
```

File: modular_skyrat/modules/shotgunrebalance/code/shotgun.dm [hunk 4]
```diff
@@ -33,13 +37,26 @@
...
+	embedding = list(
+		embed_chance = 80,
+		pain_chance = 40,
+		fall_chance = 5,
+		jostle_chance = 5,
+		ignore_throwspeed_threshold = TRUE,
+		pain_stam_pct = 0.5,
+		pain_mult = 5,
+		rip_time = 30,
+		)
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/shotgunrebalance/code/shotgun.dm]:
```suggestion
	)
```

File: modular_skyrat/modules/shotgunrebalance/code/shotgun.dm [hunk 5]
```diff
@@ -101,7 +118,10 @@
 	projectile_type = /obj/projectile/bullet/pellet/shotgun_buckshot/flechette
 	pellets = 8 //8 x 6 = 48 Damage Potential
 	variance = 25
-	custom_materials = list(/datum/material/iron=500,/datum/material/titanium=500)
+	custom_materials = list(
+		/datum/material/iron = 500,
+		/datum/material/titanium = 500,
+		)
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/shotgunrebalance/code/shotgun.dm]:
```suggestion
	)
```

File: modular_skyrat/modules/shotgunrebalance/code/shotgun.dm [hunk 6]
```diff
@@ -117,7 +137,11 @@
 	desc = "A highly experimental non-lethal shell filled with smart nanite pellets that re-aim themselves when bouncing off from surfaces. However they are not able to make out friend from foe."
 	icon_state = "cnrshell"
 	projectile_type = /obj/projectile/bullet/pellet/shotgun_buckshot/beehive
-	custom_materials = list(/datum/material/iron=500,/datum/material/silver=500,/datum/material/plasma=500)
+	custom_materials = list(
+		/datum/material/iron = 500,
+		/datum/material/silver = 500,
+		/datum/material/plasma = 500,
+		)
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/shotgunrebalance/code/shotgun.dm]:
```suggestion
	)
```

File: modular_skyrat/modules/shotgunrebalance/code/shotgun.dm [hunk 7]
```diff
@@ -147,7 +171,11 @@
 	desc = "A highly experimental shell filled with nanite electrodes that will embed themselves in soft targets. The electrodes are charged from kinetic movement which means moving targets will get punished more."
 	icon_state = "lasershell"
 	projectile_type = /obj/projectile/bullet/pellet/shotgun_buckshot/antitide
-	custom_materials = list(/datum/material/iron=500,/datum/material/gold=500,/datum/material/uranium=500)
+	custom_materials = list(
+		/datum/material/iron = 500,
+		/datum/material/gold = 500,
+		/datum/material/uranium = 500,
+		)
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/shotgunrebalance/code/shotgun.dm]:
```suggestion
	)
```

File: modular_skyrat/modules/shotgunrebalance/code/shotgun.dm [hunk 8]
```diff
@@ -168,7 +196,16 @@
...
+	embedding = list(
+		embed_chance = 70,
+		pain_chance = 25,
+		fall_chance = 15,
+		jostle_chance = 80,
+		ignore_throwspeed_threshold = TRUE,
+		pain_stam_pct = 0.9,
+		pain_mult = 2,
+		rip_time = 10,
+		)
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/shotgunrebalance/code/shotgun.dm]:
```suggestion
	)
```

File: modular_skyrat/modules/shotgunrebalance/code/shotgun.dm [hunk 9]
```diff
@@ -180,11 +217,14 @@
...
-	name = "Iceshot shell"
+	name = "Iceblox shell"
 	desc = "A highly experimental shell filled with nanites that will lower the body temperature of hit targets."
 	icon_state = "tshell"
 	projectile_type = /obj/projectile/bullet/pellet/shotgun_buckshot/iceblox
-	custom_materials = list(/datum/material/iron=500,/datum/material/plasma=500)
+	custom_materials = list(
+		/datum/material/iron = 500,
+		/datum/material/plasma = 500,
+		)
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/shotgunrebalance/code/shotgun.dm]:
```suggestion
	)
```

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]
This is strictly a code review, I didn't really take into account any sort of balance implications.

File: modular_skyrat/modules/shotgunrebalance/code/ammobox.dm (hunk 1 above)
Comment (Tastyfish) [reply modular_skyrat/modules/shotgunrebalance/code/ammobox.dm]: Taking it further, every subtype of `/obj/item/storage/box/shotgun` duplicates basically the same code. There should also be a var on ../shotgun that has the ammo type so you can just do like
```byond
for(var/i in 1 to max_ammo)
	new ammo_type(src)
```
or whatever in the shotgun/PopulateContents()

**Comment by MaslineU** [comment]
Add the ability to create these cardboard boxes, then it will be the best update.

**Comment by MaslineU** [comment]
And don't forget about the black mesa gate, and the packs of these same cartridges that are there to keep the balance.

**Comment by tf-4** [comment]
this has sat too long and got too out of date, the maps are giving me a headache. I'll fix it up on a new branch

**Pull request closed without merging**